lib/demon/sidekiq.rb

```diff
@@ -3,6 +3,57 @@
 require "demon/base"
 
 class Demon::Sidekiq < Demon::Base
+  RANDOM_HEX = SecureRandom.hex
+
+  def self.heartbeat_queues_list_key
+    @@heartbeat_queues_list_key ||= "#{RANDOM_HEX}_heartbeat_queues_list"
+  end
+
+  def self.queues_last_heartbeat_hash_key
+    @@queues_last_heartbeat_hash_key ||= "#{RANDOM_HEX}_queues_last_heartbeat_hash"
+  end
+
+  def self.trigger_heartbeat(name)
+    $redis.hset(queues_last_heartbeat_hash_key, name, Time.new.to_i.to_s)
+  end
+
+  def self.get_queue_last_heartbeat(name)
+    $redis.hget(queues_last_heartbeat_hash_key, name).to_i
+  end
+
+  def self.heartbeat_queues
+    queues = $redis.lrange(heartbeat_queues_list_key, 0, -1)
+
+    queues.select! do |queue|
+      pid = queue.split("_").last.to_i
+      alive = alive?(pid)
+      if !alive
+        $redis.lrem(heartbeat_queues_list_key, 0, queue)
+        $redis.hdel(queues_last_heartbeat_hash_key, queue)
+      end
+      alive
+    end
+    queues
+  end
+
+  def self.create_heartbeat_queue(pid)
+    queue = "#{SecureRandom.hex}_#{pid}"
+    $redis.lpush(heartbeat_queues_list_key, queue)
+    queue
+  end
+
+  def self.clear_heartbeat_queues!
+    $redis.del(heartbeat_queues_list_key)
+    $redis.del(queues_last_heartbeat_hash_key)
+  end
+
+  def self.before_start
+    # cleans up heartbeat queues from previous boot up
+    Sidekiq::Queue.all.each do |queue|
+      next if queue.name !~ /^[a-f0-9]{32}_\d+$/
+      queue.clear if queue.size == 0
```

Comment:
Shouldn't that be?

```suggestion
      queue.clear if queue.size > 0
```

Reply:
Or just

```suggestion
      queue.clear
```

Reply:
Yeah I think we can just drop this condition altogether because the queue name already matches the name of the special heartbeat queues.